ci/authn-gce/get_gce_id_tokens.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+# This script obtains all of the GCE identity tokens required by authn-gce cucumber tests
+# The scripts accepts the GCE instance name as an argument and an optional GCP zone (default is us-central1-a).
+# The script does the following:
+# 1. Validates gcloud is installed (requires a GCP account)
+#   To install gcloud goto: https://cloud.google.com/sdk/docs, download and install the SDK by running:
+#   ./google-cloud-sdk/install.sh
+#   ./google-cloud-sdk/bin/gcloud init
+# 2. Verifies the GCE instance exists and is in status running
+# 3. Executes a ssh curl command and write the output token with the appropriate token name to '../ci/authn-gce/tokens'.
+
+PROGNAME=$(basename $0)
+GCE_ZONE=us-central1-a
+TOKENS_OUT_DIR_PATH=../ci/authn-gce/tokens
+INSTANCE_EXISTS=0
+INSTANCE_RUNNING=0
+
+error_exit()
+{
+  #	----------------------------------------------------------------
+  #	Function for exit due to fatal program error
+  #		Accepts 1 argument:
+  #			string containing descriptive error message
+  #	----------------------------------------------------------------
+
+	echo "${PROGNAME}: ${1:-"Unknown Error"}" 1>&2
+	exit 1
+	echo "$1" 1>&2
+	exit 1
+}
+
+get_token_into_file() {
+  local token_format="$1"
+  local audience="$2"
+  local filename="$3"
+  local curl_cmd="curl -s -H 'Metadata-Flavor: Google' \
+    'http://metadata/computeMetadata/v1/instance/service-accounts/default/identity?format=${token_format}&audience=${audience}'"
+  echo "-- write token in ${token_format} format with audience: ${audience} to ${TOKENS_OUT_DIR_PATH}/${filename}"
+  gcloud compute ssh $INSTANCE_NAME --zone=$GCE_ZONE --command "${curl_cmd}" >"${TOKENS_OUT_DIR_PATH}/${filename}" &
+}
+
+get_tokens_into_files() {
+  if  [ "${INSTANCE_EXISTS}" = "0" ] | [ "${INSTANCE_RUNNING}" = "0" ] ; then
+    error_exit "-- Cannot run command, GCE instance '${INSTANCE_NAME}' not in valid state!"
+  else
+    get_token_into_file "full" "conjur%2Fcucumber%2Fhost%2Ftest-app" "gce_token_valid"
+    get_token_into_file "full" "conjur%2fcucumber%2fhost%2fnon-existing" "gce_token_non_existing_host"
+    get_token_into_file "full" "conjur%2fcucumber%2fhost%2fnon-rooted%2ftest-app" "gce_token_non_rooted_host"
+    get_token_into_file "full" "conjur%2fcucumber%2ftest-app" "gce_token_user"
+    get_token_into_file "full" "conjur%2fnon-existing%2fhost%2ftest-app" "gce_token_non_existing_account"
+    get_token_into_file "full" "invalid_audience" "gce_token_invalid_audience"
+    get_token_into_file "standard" "conjur%2fcucumber%2fhost%2ftest-app" "gce_token_standard_format"
+    wait
+  fi
+}
+
+check_if_gce_instance_is_running() {
+  if [ "${INSTANCE_EXISTS}" = "1" ]; then
```

Comment:
so i see that this function must run after `check_if_gce_instance_exists`. i would advise that you run this method from within that one so a future developer doesn't change the order.